### ISSUE TITLE
passcheck feature `valid_until` parameter is passing the wrong value

### ISSUE THREAD

**Issue by adamguo0** [aws/pg_tle#230]
## Description

The passcheck feature is passing an incorrect date to user functions. It seems to be passing 2000-01-01 00:00:37.560024+00.

## Steps to reproduce

1. Create and register a passcheck function that prints the `valid_until` parameter
```
psql (17devel)
Type "help" for help.

postgres=# create extension pg_tle;
CREATE EXTENSION
postgres=# create function passcheck_test(username text, password text, password_type pgtle.password_types, valid_until timestamptz, valid_null boolean)
returns void as $$
begin
raise exception '%', valid_until;
end
$$ language plpgsql;
CREATE FUNCTION
postgres=# select pgtle.register_feature('passcheck_test', 'passcheck');
 register_feature
------------------

(1 row)
```

2. Alter a role's password
```
postgres=# alter role adamguo password 'pass' valid until '2023-12-31';
ERROR:  2000-01-01 00:00:37.560024+00
CONTEXT:  PL/pgSQL function passcheck_test(text,text,pgtle.password_types,timestamp with time zone,boolean) line 3 at RAISE
SQL statement "SELECT public.passcheck_test($1::pg_catalog.text, $2::pg_catalog.text, $3::pgtle.password_types, $4::pg_catalog.timestamptz, $5::pg_catalog.bool)"
```

## Expected outcome

The error should read `ERROR:  2023-12-31 00:00:00.000000+00`, which is the given valid until date.

## Actual outcome

`ERROR:  2000-01-01 00:00:37.560024+00`, which is the wrong date.

## Analysis

In [passcheck.c](https://github.com/aws/pg_tle/blob/main/src/passcheck.c#L160), we use *GetDatum function calls to get the other four parameters, but we use DirectionFunctionCall1 to timestamptz_out for the valid_until parameter. We can try using TimestampTzGetDatum instead.

We are also missing unit tests for valid_until and valid_null.

**Comment by rogerio-satler-deel** [comment]
Hi,
When running the following: 
`create user test WITH PASSWORD 'xxx' VALID UNTIL '2025-03-02';`
The value on `valid_until` variable on the parameter is `2000-09-17`
I'm still getting this same issue using the installation as:
```
SELECT pgtle.install_extension (
  'password_check_rules',
  '1.0',
  'Verify Users passwords'..
```
  Also with:
```
  SELECT pgtle.install_extension (
  'password_check_rules',
  '1.3.3', --same on 1.2.0
  'Verify Users passwords'..
```
  
  When this fix will be available?

**Comment by adamguo0** [comment]
Hi @rogerio-satler-deel, which version of pg_tle are you testing this on?

**Comment by rogerio-satler-deel** [comment]
> Hi @rogerio-satler-deel, which version of pg_tle are you testing this on?

I tested on 1.0, 1.2.0 and 1.3.3 on PostgreSQL 15

**Comment by adamguo0** [comment]
Thanks, I noticed in your example that you're writing the versions 1.0.0, 1.2.0, and 1.3.3 in the `pgtle.install_extension` call. Those versions determine the versions of your trusted language extension, not the version of pg_tle used. Can you post the output of `\dx` in psql?

The fix for this issue should be available starting in pg_tle 1.3.0.

**Comment by adamguo0** [comment]
You can compile and update pg_tle by following the [installation instructions](https://github.com/aws/pg_tle/blob/main/docs/01_install.md) if you are self-managing Postgres.
